src/irmin/store.ml

```diff
@@ -903,6 +903,54 @@ module Make (P: S.PRIVATE) = struct
         | _ -> Lwt.return_none
       ) g
 
+  module Heap = Bheap.Make(struct
+      type t = commit * int
+      let compare c1 c2 =
+         Int64.compare
+          (Info.date (Commit.info (fst c1)))
+          (Info.date (Commit.info (fst c2)))
+    end)
+
+  let last_modified ?depth ?(number = 1) t key =
+    Head.get t >>= fun commit ->
+    history t >>= fun history ->
```

Comment:
This should not be there. `history t` will explicitly compute the graph of commits which can be rather large.